include/mapbox/variant.hpp

```diff
@@ -861,6 +862,22 @@ class variant
         return detail::binary_dispatcher<F, V, R, Types...>::apply(v0, v1, std::forward<F>(f));
     }
 
+    // match
+    // unary
+    template <typename... Fs>
+    auto VARIANT_INLINE match(Fs&&... fs) const
+        -> decltype(detail::dispatcher<typename ::mapbox::util::visitor<Fs...>::type, variant, typename detail::result_of_unary_visit<typename ::mapbox::util::visitor<Fs...>::type, first_type>::type, Types...>::apply_const(*this, ::mapbox::util::make_visitor(std::forward<Fs>(fs)...)))
+    {
+        return detail::dispatcher<typename ::mapbox::util::visitor<Fs...>::type, variant, typename detail::result_of_unary_visit<typename ::mapbox::util::visitor<Fs...>::type, first_type>::type, Types...>::apply_const(*this, ::mapbox::util::make_visitor(std::forward<Fs>(fs)...));
+    }
+    // non-const
+    template <typename... Fs>
+    auto VARIANT_INLINE match(Fs&&... fs)
+        -> decltype(detail::dispatcher<typename ::mapbox::util::visitor<Fs...>::type, variant, typename detail::result_of_unary_visit<typename ::mapbox::util::visitor<Fs...>::type, first_type>::type, Types...>::apply(*this, ::mapbox::util::make_visitor(std::forward<Fs>(fs)...)))
+    {
+        return detail::dispatcher<typename ::mapbox::util::visitor<Fs...>::type, variant, typename detail::result_of_unary_visit<typename ::mapbox::util::visitor<Fs...>::type, first_type>::type, Types...>::apply(*this, ::mapbox::util::make_visitor(std::forward<Fs>(fs)...));
+    }
```

Comment:
Um, actually the easiest thing you can do is call `visit` instead of duplicating it here. That'll hide all the dispatch ugliness.